isthmus/src/test/java/io/substrait/isthmus/Substrait2SqlTest.java

```diff
@@ -42,7 +42,78 @@ public void simpleTestDecimal() throws Exception {
         "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0 from lineitem where l_shipdate < date '1998-01-01' ");
   }
 
+  @Test
+  public void simpleJoin() throws Exception {
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem left join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem right join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+    test(
+        "select l_partkey + l_orderkey, l_extendedprice * 0.1 + 100.0, o_orderkey from lineitem full join orders on l_orderkey = o_orderkey where l_shipdate < date '1998-01-01' ");
+  }
+
+  @Test
+  public void simpleTestAgg() throws Exception {
+    test(
+        "select l_partkey, count(l_tax), COUNT(distinct l_discount) from lineitem group by l_partkey");
+  }
+
+  @Test
+  public void simpleTestAggNoGB() throws Exception {
+    test("select count(l_tax), count(distinct l_discount) from lineitem");
+  }
+
+  @Test
+  public void simpleTestAgg2() throws Exception {
+    test(
+        "select l_partkey, sum(l_tax), sum(distinct l_tax), avg(l_discount), avg(distinct l_discount) from lineitem group by l_partkey");
+  }
+
+  @Test
+  public void simpleTestAgg3() throws Exception {
+    // do not verify pojo in round trip test, because of implicit cast(1 as decimal) is removed in
+    // round trip
+    // conversion.
+    test(
+        "select l_partkey, sum(l_extendedprice * (1 - l_discount)) from lineitem group by l_partkey",
+        false);
+  }
+
+  @Test
+  public void tpch_q1_variant() throws Exception {
+    // difference from tpch_q1 : 1) remove order by clause; 2) remove interval date literal
+    // do not verify pojo in round trip test, because of implicit cast(1 as decimal) is removed in
+    // round trip
+    // conversion.
+    test(
+        "select\n"
+            + "  l_returnflag,\n"
+            + "  l_linestatus,\n"
+            + "  sum(l_quantity) as sum_qty,\n"
+            + "  sum(l_extendedprice) as sum_base_price,\n"
+            + "  sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,\n"
```

Comment:
If you use `1.0` instead of `1` or if you add the cast to decimal, does the roundtrip equality succeed?

Reply:
good idea. Using 1.0 will avoid the implicit cast difference in the round trip conversion.